lex.go

```diff
@@ -43,14 +43,60 @@ var tokChanCap = TokenChanCap // capacity of lexer token channel
 func SetTokenChanCap(size int) {
 	tokChanCap = size
 }
-func Lex(reader io.Reader) chan NgxToken {
+
+// ExternalLexer is an interface for implementing lexers that handle external NGINX tokens during the lexical analysis phase.
+type ExternalLexer interface {
```

Comment:
Up above I suggested removing `External` from the interface name. I think the names look better and more "Go-like" without `External` being part of the interface names, but I see how that feedback can be hard to apply here. We can sync up and figure out if we can think of better names or if we want to just stick with the ones that exist.

Reply:
Removed the` External`.